onboarding/main.go

```diff
@@ -0,0 +1,138 @@
+	onboardingTicketPrivateKeySecretName = "onboarding-ticket-private-key"
+	serviceAccountName                   = "onboarding-secret-generator"
+)
+
+func main() {
+
+	clientset, err := newClient()
+	if err != nil {
+		klog.Error(err, "failed to create controller-runtime client")
+		return
+	}
+
+	operatorNamespace, err := util.GetOperatorNamespace()
+	if err != nil {
+		klog.Error(err, "unable to get operator namespace")
+		os.Exit(1)
+	}
+
+	// 1. Check public key secret exist or not
+	_, err = clientset.CoreV1().Secrets(operatorNamespace).Get(context.TODO(), onboardingTicketPublicKeySecretName, metav1.GetOptions{})
+
+	if err != nil && kerrors.IsNotFound(err) {
+
+		// Generate RSA key.
+		var err error
+		privateKey, err := rsa.GenerateKey(rand.Reader, 4096)
+		if err != nil {
+			panic(err)
+		}
+
+		publicKey := &privateKey.PublicKey
+
+		// Export the keys to pem string
+		privatePem := convertRsaPrivateKeyAsPemStr(privateKey)
+		publicPem, err := convertRsaPublicKeyAsPemStr(publicKey)
+
+		if err != nil {
+			panic(err)
+		}
+
+		privateSecret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:        onboardingTicketPrivateKeySecretName,
+				Namespace:   operatorNamespace,
+				Annotations: map[string]string{"kubernetes.io/service-account.name": serviceAccountName},
+			},
+			Type: "kubernetes.io/service-account-token",
+			StringData: map[string]string{
+				"key": privatePem,
+			},
+		}
+
+		_, err = clientset.CoreV1().Secrets(operatorNamespace).Create(context.Background(), privateSecret, metav1.CreateOptions{})
+
+		if err != nil {
+			klog.Error(err, "Failed to create secret.")
+		}
+
+		publicSecret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      onboardingTicketPublicKeySecretName,
+				Namespace: operatorNamespace,
+			},
+			StringData: map[string]string{
+				"key": publicPem,
+			},
+		}
+
+		_, err = clientset.CoreV1().Secrets(operatorNamespace).Create(context.Background(), publicSecret, metav1.CreateOptions{})
+
+		if err != nil {
+			klog.Error(err, "Failed to create secret.")
+		}
+
+	}
+
+}
+
+func newClient() (*kubernetes.Clientset, error) {
+
+	config := runtime.GetConfigOrDie()
+
+	clientset, err := kubernetes.NewForConfig(config)
+
+	if err != nil {
+		klog.Error(err, "failed to get clientset")
+
+	}
+	return clientset, nil
+
+}
+
+func convertRsaPrivateKeyAsPemStr(privateKey *rsa.PrivateKey) string {
+	privteKeyBytes := x509.MarshalPKCS1PrivateKey(privateKey)
+	privateKeyPem := pem.EncodeToMemory(
+		&pem.Block{
+			Type:  "RSA PRIVATE KEY",
+			Bytes: privteKeyBytes,
+		},
+	)
+	return string(privateKeyPem)
+}
+
+func convertRsaPublicKeyAsPemStr(publicKey *rsa.PublicKey) (string, error) {
+	publicKeyBytes, err := x509.MarshalPKIXPublicKey(publicKey)
+	if err != nil {
+		return "", err
+	}
+	publicKeyPem := pem.EncodeToMemory(
+		&pem.Block{
+			Type:  "RSA PUBLIC KEY",
+			Bytes: publicKeyBytes,
+		},
+	)
+
+	return string(publicKeyPem), nil
+}
```

Comment:
all these 3 functions can be inlined.

Reply:
Difficult to review and read if i do this entire function single line